specs/ProcessInfo.md

```diff
@@ -0,0 +1,278 @@
+  /// Indicates the PPAPI plugin broker process kind.
+  COREWEBVIEW2_PROCESS_KIND_PPAPI_BROKER,
+
+  /// Indicates the process of unspecified kind.
+  COREWEBVIEW2_PROCESS_KIND_UNKNOWN,
+} COREWEBVIEW2_PROCESS_KIND;
+
+[uuid(20856F87-256B-41BE-BD64-AB1C36E3D944), object, pointer_default(unique)]
+interface ICoreWebView2Environment6 : ICoreWebView2Environment5
+{
+  /// Adds an event handler for the `ProcessInfoChanged` event.
+  /// 
+  /// \snippet ProcessComponent.cpp ProcessInfoChanged
+  HRESULT add_ProcessInfoChanged(
+      [in] ICoreWebView2ProcessInfoChangedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ProcessInfoChanged`.
+  HRESULT remove_ProcessInfoChanged(
+      [in] EventRegistrationToken token);
+
+  /// Returns the `ICoreWebView2ProcessInfoCollection`
+  /// Provide a list of all process using same user data folder.
+  [propget] HRESULT ProcessInfo([out, retval]ICoreWebView2ProcessInfoCollection** value);
+}
+
+/// Provides a set of properties for a process in the `ICoreWebView2Environment`.
+[uuid(7798D399-52A1-4823-AD6A-1F3EDD74B0B6), object, pointer_default(unique)]
+interface ICoreWebView2ProcessInfo : IUnknown {
+
+  /// The process id of the process.
+  [propget] HRESULT Id([out, retval] UINT32* value);
+
+  /// The kind of the process.
+  [propget] HRESULT Kind([out, retval] COREWEBVIEW2_PROCESS_KIND* kind);
+}
+
+/// A list containing process id and corresponding process type.
+[uuid(5356F3B3-4859-4763-9C95-837CDEEE8912), object, pointer_default(unique)]
+interface ICoreWebView2ProcessInfoCollection : IUnknown {
+  /// The number of process contained in the ICoreWebView2ProcessInfoCollection.
+  [propget] HRESULT Count([out, retval] UINT* count);
+
+  /// Gets the `ICoreWebView2ProcessInfo` located in the `ICoreWebView2ProcessInfoCollection`
+  /// at the given index.
+  HRESULT GetValueAtIndex([in] UINT32 index,
+                          [out, retval] ICoreWebView2ProcessInfo** processInfo);
+}
+
+/// An event handler for the `ProcessInfoChanged` event.
+[uuid(CFF13C72-2E3B-4812-96FB-DFDDE67FBE90), object, pointer_default(unique)]
+interface ICoreWebView2ProcessInfoChangedEventHandler : IUnknown {
+  /// Provides the event args for the corresponding event.  No event args exist
+  /// and the `args` parameter is set to `null`.
+  HRESULT Invoke([in] ICoreWebView2* sender, [in] IUnknown* args);
+}
+```
+
+```c# (but really MIDL3)
+namespace Microsoft.Web.WebView2.Core
+{
+    // ...
+    runtimeclass CoreWebView2ProcessInfoCollection;
+    runtimeclass CoreWebView2ProcessInfo;
+
+    /// Kind of process type used in the CoreWebView2ProcessCollection.
+    enum CoreWebView2ProcessKind
+    {
+        Browser = 0,
+        Renderer = 1,
+        Utility = 2,
+        SandboxHelper = 3,
+        Gpu = 4,
+        PpapiPlugin = 5,
+        PpapiBroker = 6,
+        Unknown = 7,
+    };
+
+    runtimeclass CoreWebView2ProcessInfoCollection
```

Comment:
This runtimeclass shouldn't exist because you are using IVectorView below as your collection of ProcessInfo objects

Reply:
You should be able to skip this class in the winrt & .NET stgs since its not used at all

Reply:
Right but it shouldn't be there. You can add it to the list of things to skip in the .NET stg and WinRT IDL stg files.